### DISCUSSION TITLE
Migrating from docker to podman and caching problem

### DISCUSSION THREAD

**Discussion by gilgameshfreedom** [containers/podman#18185] [Q&A] [answered]
Hey all!

I'd like to migrate from docker to podman to build images. In some pipelines we use `--cache-from` option in docker to reuse layers from the previous built images. It looks like that:

1. docker pull imageA:latest
2. docker build --cache-from imageA:latest -t imageB:latest .
3. docker push imageB:latest

We use `--cache-from` **not** to decrease build's time, but to provide some kind of "idempotency" - it guarantees that the imageB has the same versions of the packages (the versions of the packages are not pined in the RUN step) if the RUN step which install packages hasn't changed.

Unfortunately if I just replace docker to podman it won't work because podman (buildah) uses another caching mechanism - https://github.com/containers/buildah/pull/4118, so the first building with podman rebuild all the layers (and it obviously will update package's versions), because there's no cache in the repo.

Does anyone know how I can migrate from docker to podman without rebuilding all the layers?

**Comment by flouthoc** [comment, marked as answer]
Hi @gilgameshfreedom , podman and buildah's distributed cache is by design like this, it works with intermediate images not with an existing pre-built image. It pushesh all intermediate images to remote repo and uses the required layers during the build from remote repo. Therefore rebuilding and populating the upstream repo with all intermediate images is the only way for now.

TLDR: Docker's inbuilt distributed cache cannot be used with podman/buildah as now since it uses different internal design and mechanism.

Thanks
